src/measures/MultiS3.cpp

```diff
@@ -1,29 +1,53 @@
 #include "MultiS3.hpp"
 #include <cmath>
 
+unsigned NUM_GRAPHS;
 unsigned MultiS3::denom = 0;
-vector<uint> MultiS3::shadowDegrees;
+vector<uint> MultiS3::totalDegrees;
 
 MultiS3::MultiS3(Graph* G1, Graph* G2) : Measure(G1, G2, "ms3")
 {
+    extern char *getetv(char*);
+    char *s = getenv((char*)"NUM_GRAPHS");
+    if (s)
+    {
+        assert(1 == sscanf(s, "%u",&NUM_GRAPHS));
+    }
+    else
+    {
+        cout << "Warning: NUM_GRAPHS should be an environment variables; setting to 2 for now\n";
+        NUM_GRAPHS = 2;
+    }
+    cout << "Multi S3: NUM_GRAPHS = " << NUM_GRAPHS << endl;
+    degreesInit = false;
     //G1->printStats(0, cout);
     //G2->printStats(0, cout);
-    initDegrees(*G2);
 }
 
 MultiS3::~MultiS3() {
 }
 
-void MultiS3::initDegrees(const Graph& G2)
+void MultiS3::initDegrees(const Alignment& A, const Graph& G1, const Graph& G2)
 {
-    shadowDegrees = vector<uint>(G2.getNumNodes(), 0);
+    totalDegrees = vector<uint>(G2.getNumNodes(), 0);
+    
+    vector<vector<uint> > G1EdgeList;
+    G1.getEdgeList(G1EdgeList);
     
     vector<vector<uint>> G2AdjLists;
     G2.getAdjLists(G2AdjLists);
     
     Matrix<MATRIX_UNIT> G2Matrix;
     G2.getMatrix(G2Matrix);
     
+    uint node1, node2;
+
+    for (const auto& edge: G1EdgeList)

```

Comment:
no, we simply want to loop over each *node* in G1, touching each node exactly once. We don't loop through the *edges*, because this'll touch each node many times.

Reply:
I'm almost certain the loop over edges is wrong, because totalDegree of a shadow node has to include also pairs of nodes that are *not* connected in G1. So we have to loop over *all* pairs of nods in G1, not just G1's edges.  I'll accept the PR now and give it a try, and it may not crash, but it'll be optimizing the wrong objective. :-)

Reply:
Good point, this one I will fix first